### PR TITLE
Fix an imperfect implementation of tf.losses.mean_pairwise_squared_error

### PR DESCRIPTION
**Pull request by chofchof**
Here is a fix for the issue [Imperfect implementation of tf.losses.mean_pairwise_squared_error (#15968)](https://github.com/tensorflow/tensorflow/issues/15968)

RELNOTES: Fixed wrong normalization in tf.losses.mean_pairwise_squared_error to conform to the math and documentation. Numerical results will be different.

### CODE REVIEW

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
Thank you!

**Comment by rmlarsen** [comment]
@chofchof Can you please fix the broken unit tests?

https://source.cloud.google.com/results/invocations/347190b4-ca0d-4a73-9e53-968668852429/targets/%2F%2Ftensorflow%2Fpython%2Fkernel_tests:losses_test/log

**Comment by chofchof** [comment]
@rmlarsen Do you want for me to open a new "Pull request"? To pass the test, `tensorflow/python/kernel_tests/losses_test.py` needs to be fixed as follows:

    --- losses_test.py.orig 2018-02-01 18:10:54.865671816 +0900
    +++ losses_test.py      2018-02-01 18:05:41.668652970 +0900
    @@ -953,14 +953,14 @@
         # Compute the expected loss 'manually'.
         total = np.zeros((batch_size,))
         for b in range(batch_size):
    -      for i in range(dims):
    -        for j in range(dims):
    +      for i in range(dims-1):
    +        for j in range(i+1, dims):
               x = self._predictions[b, i].item() - self._predictions[b, j].item()
               y = self._labels[b, i].item() - self._labels[b, j].item()
               diff = (x - y)
               total[b] += (diff * diff)

    -    self._expected_losses = np.divide(total, 9.0)
    +    self._expected_losses = np.divide(total, 3.0)

       def testValueErrorThrownWhenWeightIsNone(self):
         with self.test_session():
    @@ -1060,7 +1060,7 @@
             [[8, 1, 3], [7, 8, 9], [10, 11, 12]],
         ])
         self._test_valid_weights(
    -        labels, predictions, expected_loss=122.22222)
    +        labels, predictions, expected_loss=137.5)

       def test3dWeightedScalar(self):
         labels = np.array([
    @@ -1073,7 +1073,7 @@
         ])
         weight = 3.0
         self._test_valid_weights(
    -        labels, predictions, expected_loss=weight * 122.22222,
    +        labels, predictions, expected_loss=weight * 137.5,
             weights=weight)

       def _test_invalid_weights(
    @@ -1124,7 +1124,7 @@
         ])
         self._test_valid_weights(
             # TODO(ptucker): This doesn't look right.
    -        labels, predictions, expected_loss=9 * 122.22222,
    +        labels, predictions, expected_loss=9 * 137.5,
             weights=np.ones((2, 3, 3)))

       def testLossWithAllZeroBatchSpecificWeights(self):

**Comment by rmlarsen** [comment]
@chofchof please update the broken unit tests.

**Comment by rmlarsen** [comment]
@chofchof Thanks for the contribution!
